turbopack/crates/turbo-tasks-backend/src/backend/storage.rs

```diff
@@ -403,15 +407,45 @@ macro_rules! get {
     };
 }
 
+macro_rules! get_mut {
+    ($task:ident, $key:ident $input:tt) => {
+        if let Some($crate::data::CachedDataItemValue::$key { value }) = $task.get_mut(&$crate::data::CachedDataItemKey::$key $input).as_mut() {
+            let () = $crate::data::allow_mut_access::$key;
+            Some(value)
+        } else {
+            None
+        }
+    };
+    ($task:ident, $key:ident) => {
+        $crate::backend::storage::get_mut!($task, $key {})
+    };
+}
+
 macro_rules! iter_many {
-    ($task:ident, $key:ident $input:tt => $value:ident) => {
+    ($task:ident, $key:ident $input:tt => $value:expr) => {
         $task
             .iter($crate::data::indicies::$key)
             .filter_map(|(key, _)| match *key {
                 $crate::data::CachedDataItemKey::$key $input => Some($value),
                 _ => None,
             })
     };
+    ($task:ident, $key:ident $input:tt => $value:expr) => {
+        $task
+            .iter($crate::data::indicies::$key)
+            .filter_map(|(key, _)| match key {
+                $crate::data::CachedDataItemKey::$key $input => Some($value),
+                _ => None,
+            })
+    };
+    ($task:ident, $key:ident $input:tt if $cond:expr => $value:expr) => {
+        $task
+            .iter($crate::data::indicies::$key)
+            .filter_map(|(key, _)| match key {
+                $crate::data::CachedDataItemKey::$key $input if $cond => Some($value),
+                _ => None,
+            })
+    };
```

Comment:
The introduces inconsistency with how derefs happen, as the first branch uses `*key`, but this doesn't. I don't care too much what we do, but it has to be consistent.

Should be fixed in #71061